### PR TITLE
Shapeit5 ligate

### PR DESCRIPTION
**Pull request by LouisLeNezet**
## PR checklist

New Shapeit 5 module: Ligate
Some improvement done on Shapeit 5 phase common for better identification of output files.

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/shapeit5/ligate/main.nf [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    tag "$meta.id"
+    label 'process_low'
+
+    conda "bioconda::shapeit5=1.0.0"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/shapeit5:1.0.0--h0c8ee15_0':
+        'quay.io/biocontainers/shapeit5:1.0.0--h0c8ee15_0'}"
+
+    input:
+    tuple val(meta), path(input_list, stageAs: 'dir*/*'), path (input_list_index, stageAs: 'dir*/*')
```
Comment (nvnieuwk) [inline modules/nf-core/shapeit5/ligate/main.nf:11]: some people don't like the `stageAs`. Don't count me as one of them, just saying that you can get some comments on it later (they usually prefer that all input files have different names)

File: modules/nf-core/shapeit5/ligate/main.nf [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        --input all_files.txt \\
+        --thread $task.cpus \\
+        --output ${prefix}.${suffix}
+
+    cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            shapeit5: "\$(SHAPEIT5_ligate | sed -nr '/Version/p' | grep -o -E '([0-9]+.){1,2}[0-9]' | head -n 1)"
+    END_VERSIONS
+    """
+}
```
Comment (nvnieuwk) [inline modules/nf-core/shapeit5/ligate/main.nf:38]: Can you add stub?

File: modules/nf-core/shapeit5/ligate/meta.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
+name: "shapeit5_ligate"
+description: write your description here
```
Comment (nvnieuwk) [inline modules/nf-core/shapeit5/ligate/meta.yml:2]: Oops not description here 😁

File: modules/nf-core/shapeit5/ligate/meta.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+input:
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. [ id:'test', single_end:false ]
+
+  - input_list:
+      type: file
+      description: VCF/BCF file containing genotype probabilities (GP field).
```
Comment (nvnieuwk) [inline modules/nf-core/shapeit5/ligate/meta.yml:25]:
```suggestion
      description: VCF/BCF files containing genotype probabilities (GP field).
```

**Review by nvnieuwk — COMMENTED** [review]
Some small comments :)

File: modules/nf-core/shapeit5/ligate/main.nf (hunk 1 above)
Comment (LouisLeNezet) [reply modules/nf-core/shapeit5/ligate/main.nf:11]: I understand completely I will do another way!

File: modules/nf-core/shapeit5/ligate/meta.yml (hunk 3 above)
Comment (LouisLeNezet) [reply modules/nf-core/shapeit5/ligate/meta.yml:2]: Sorry I forgot.

Comment (nvnieuwk) [reply modules/nf-core/shapeit5/ligate/meta.yml:2]: No problem 😛

File: modules/nf-core/shapeit5/ligate/meta.yml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+      description: |
+        Groovy Map containing sample information
+        e.g. [ id:'test', single_end:false ]
+
+  - input_list:
+      type: file
+      description: VCF/BCF files containing genotype probabilities (GP field).
+      pattern: "*.{vcf,bcf,vcf.gz,bcf.gz}"
+
+  - input_list:
```
Comment (nvnieuwk) [inline modules/nf-core/shapeit5/ligate/meta.yml:30]:
```suggestion
  - input_list_index:
```

**Comment by LouisLeNezet** [comment]
For a reason that I do not  yet fully understand, sometimes everything works fine and sometimes Shapeit5 ligate give a `Segmentation fault`.
I will be happy to have some help on that.
I will create an issue on the creator github.

**Comment by nvnieuwk** [comment]
> For a reason that I do not yet fully understand, sometimes everything works fine and sometimes Shapeit5 ligate give a `Segmentation fault`. I will be happy to have some help on that. I will create an issue on the creator github.

That's strange, but probably an issue with the tool itself (as everything looks fine on the nextflow side). Let's hope they quickly respond on your issue

**Comment by LouisLeNezet** [comment]
I found the problem.
It was just that files wasn't ordered correctly.

**Review by nvnieuwk — APPROVED** [review]

**Comment by LouisLeNezet** [comment]
Thanks @nvnieuwk !

**Pull request merged**